rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -0,0 +1,357 @@
+
+vx_enum vx_mem_type(RocalMemType mem)
+{
+    switch(mem)
+    {
+        case RocalMemType::OCL:
+        {
+            return VX_MEMORY_TYPE_OPENCL;
+        }
+        break;
+        case RocalMemType::HOST:
+        {
+            return VX_MEMORY_TYPE_HOST;
+        }
+        break;
+        case RocalMemType::HIP:
+        {
+            return VX_MEMORY_TYPE_HIP;
+        }
+        break;
+        default:
+            throw std::runtime_error("Memory type not valid");
+    }
+}
+
+vx_size tensor_data_size(RocalTensorDataType data_type)
+{
+    switch(data_type)
+    {
+        case RocalTensorDataType::FP32:
+        {
+            return sizeof(vx_float32);
+        }
+        break;
+        case RocalTensorDataType::FP16:
+        {
+            return sizeof(vx_int16);
+        }
+        break;
+        case RocalTensorDataType::UINT8:
+        {
+            return sizeof(vx_uint8);
+        }
+        case RocalTensorDataType::UINT32:
+        {
+            return sizeof(vx_uint32);
+        }
+        case RocalTensorDataType::INT32:
+        {
+            return sizeof(vx_int32);
+        }
+        break;
+        default:
+            throw std::runtime_error("tensor data_type not valid");
+    }
+}
+
+bool operator==(const rocALTensorInfo &rhs, const rocALTensorInfo &lhs)
+{
+    return (rhs.dims() == lhs.dims() &&
+            rhs.mem_type() == lhs.mem_type() &&
+            rhs.data_type() == lhs.data_type() &&
+            rhs.color_format() == lhs.color_format() &&
+            rhs.layout() == lhs.layout());
+}
+
+void rocALTensorInfo::reallocate_tensor_roi_buffers()
+{
+    _roi = std::make_shared<std::vector<RocalROI>>(_batch_size);
+
+    if(_roi->size() > 0)
+        _roi->clear();
+    _roi->resize(_batch_size);
+    if(_is_image)
+    {
+        for (unsigned i = 0; i < _batch_size; i++)
+        {
+            _roi->at(i).x1 = 0;
+            _roi->at(i).y1 = 0;
+            _roi->at(i).x2 = _max_dims.at(0);
+            _roi->at(i).y2 = _max_dims.at(1);
+        }
+    }
+    else
+    {
+        // TODO - For other tensor types
+    }
+}
+
+rocALTensorInfo::rocALTensorInfo() : _type(Type::UNKNOWN),
+                                     _num_of_dims(0),
+                                     _dims({}),
+                                     _mem_type(RocalMemType::HOST),
+                                     _data_type(RocalTensorDataType::FP32){}
+
+rocALTensorInfo::rocALTensorInfo(
+    std::vector<size_t> dims,
+    RocalMemType mem_type,
+    RocalTensorDataType data_type) : _type(Type::UNKNOWN),
+                                     _dims(dims),
+                                     _mem_type(mem_type),
+                                     _data_type(data_type)
+{
+    _batch_size = dims.at(0);
+    _num_of_dims = dims.size();
+    _data_size = tensor_data_size(data_type);
+    for(unsigned i = 0; i < _num_of_dims; i++)
+        _data_size *= dims.at(i);
+
+    if(_num_of_dims <= 3)
```

Comment:
I am not sure if this is a correct statement

Reply:
The image tensors will always be of 4 dimension or greater considering video sequences also(i.e NCHW, NHWC, NFHWC, NFCHW). Hence if the num of dims is less than 3 we are setting is_image flag to false